cmd/list.go

```diff
@@ -82,57 +98,141 @@ func list(cmd *cobra.Command) {
 		token = os.Getenv("REANA_ACCESS_TOKEN")
 	}
 	workflow, _ := cmd.Flags().GetString("workflow")
+	listAllSessions, _ := cmd.Flags().GetBool("sessions")
 	jsonOutput, _ := cmd.Flags().GetBool("json")
-	filter, _ := cmd.Flags().GetStringArray("filter")
+	showAll, _ := cmd.Flags().GetBool("all")
+	verbose, _ := cmd.Flags().GetBool("verbose")
+	humanReadable, _ := cmd.Flags().GetBool("human-readable")
+	filters, _ := cmd.Flags().GetStringArray("filter")
+	includeDuration, _ := cmd.Flags().GetBool("include-duration")
+	includeProgress, _ := cmd.Flags().GetBool("include-progress")
+	includeWorkspaceSize, _ := cmd.Flags().GetBool("include-workspace-size")
+	showDeletedRuns, _ := cmd.Flags().GetBool("show-deleted-runs")
+	page, _ := cmd.Flags().GetInt64("page")
+	size, _ := cmd.Flags().GetInt64("size")
 
-	filterNames := []string{"name", "status"}
-	statusFilters, searchFilter := utils.ParseListFilters(filter, filterNames)
+	var runType string
+	if listAllSessions {
+		runType = "interactive"
+	} else {
+		runType = "batch"
+	}
+
+	statusFilters := utils.GetRunStatuses(showDeletedRuns || showAll)
+	var searchFilter string
+	if len(filters) > 0 {
+		filterNames := []string{"name", "status"}
+		statusFilters, searchFilter = utils.ParseListFilters(filters, filterNames)
+	}
 
 	listParams := operations.NewGetWorkflowsParams()
 	listParams.SetAccessToken(&token)
+	listParams.SetType(runType)
+	listParams.SetVerbose(&verbose)
+	listParams.SetPage(&page)
 	listParams.SetWorkflowIDOrName(&workflow)
 	listParams.SetStatus(statusFilters)
 	listParams.SetSearch(&searchFilter)
+	listParams.SetIncludeProgress(&includeProgress)
+	listParams.SetIncludeWorkspaceSize(&includeWorkspaceSize)
+	if size > 0 {
+		listParams.SetSize(&size)
+	}
 
 	listResp, err := client.ApiClient().Operations.GetWorkflows(listParams)
 	if err != nil {
 		fmt.Println("Error: ", err)
 		os.Exit(1)
 	}
+
+	headers := buildListHeaders(
+		runType,
+		verbose,
+		includeWorkspaceSize,
+		includeProgress,
+		includeDuration,
+	)
+	displayListPayload(listResp.Payload, headers[runType], jsonOutput, humanReadable)
+}
+
+func displayListPayload(
+	p *operations.GetWorkflowsOKBody,
+	header []any,
+	jsonOutput, humanReadable bool,
+) {
+	var data [][]any
+	for _, workflow := range p.Items {
+		name, runNumber := getWorkflowNameAndRunNumber(workflow.Name)
+
+		var row []any
+		for _, col := range header {
+			var value any
+
+			switch col {
+			case "id":
+				value = workflow.ID
+			case "user":
+				value = workflow.User
+			case "size":
+				if humanReadable {
+					value = workflow.Size.HumanReadable
+				} else {
+					value = workflow.Size.Raw
+				}
+			case "progress":
+				// TODO check if I have to be careful with nils
```

Comment:
Do you know if it's possible for this information to be missing? The Python version checked for nulls but I'm not sure and it'll make the code more confusing

Reply:
I noticed it actually comes as 0 (GO's default value) so I'll try to manage something